mws/apis/subscriptions.py

```diff
@@ -17,76 +18,145 @@ class Subscriptions(MWS):
 
         Docs:
         http://docs.developer.amazonservices.com/en_US/subscriptions/Subscriptions_DeregisterDestination.html
         """
-        raise NotImplementedError
+        if attribute_list is None:
+            raise ValueError("Attribute_list cannot be None")
+
+        data = {
+            "Action": "DeregisterDestination",
+            "MarketplaceId": marketplace_id,
+            "Destination.DeliveryChannel": delivery_channel
+        }
+        data.update(utils.enumerate_keyed_param("Destination.AttributeList.member", attribute_list))
 
-    def list_registered_destinations(self):
+        return self.make_request(data, "POST")
+
+    def list_registered_destinations(self, marketplace_id):
         """
         Lists all current destinations that you have registered.
 
         Docs:
         http://docs.developer.amazonservices.com/en_US/subscriptions/Subscriptions_ListRegisteredDestinations.html
         """
-        raise NotImplementedError
+        data = {"Action": "ListRegisteredDestinations",
+                "MarketplaceId": marketplace_id}
+
+        return self.make_request(data, "POST")
 
-    def send_test_notification_to_destination(self):
+    def send_test_notification_to_destination(self, marketplace_id, delivery_channel="SQS", attribute_list=None):
         """
         Sends a test notification to an existing destination.
 
         Docs:
         http://docs.developer.amazonservices.com/en_US/subscriptions/Subscriptions_SendTestNotificationToDestination.html
+
+        destination_list ={ "Key": "sqsQueueUrl",
+         "Value": "destination" }
         """
-        raise NotImplementedError
+        if attribute_list is None:
+            raise ValueError("Attribute_list cannot be None")
+
+        data = {"Action": "SendTestNotificationToDestination",
+                "MarketplaceId": marketplace_id,
+                "Destination.DeliveryChannel": delivery_channel}
+        data.update(utils.enumerate_keyed_param("Destination.AttributeList.member", attribute_list))
+
+        return self.make_request(data, method="POST")
 
-    def create_subscription(self):
+    def create_subscription(self, marketplace_id, delivery_channel="SQS", attribute_list=None,
+                            _type=None, is_enabled=True):
         """
         Creates a new subscription for the specified notification type and destination.
 
         Docs:
         http://docs.developer.amazonservices.com/en_US/subscriptions/Subscriptions_CreateSubscription.html
         """
-        raise NotImplementedError
+        if attribute_list is None:
+            raise ValueError("Attribute_list cannot be None")
+        if _type is None:
+            raise ValueError("_type cannot be None")
+        data = {"Action": "CreateSubscription",
+                "MarketplaceId": marketplace_id,
+                "Subscription.Destination.DeliveryChannel": delivery_channel,
+                "Subscription.IsEnabled": str(is_enabled).lower(),
```

Comment:
Other active PR is working on a "clean" method in which we should be able to pass bool literals moving forward, so `str(is_enabled).lower()` will not strictly be necessary later on. This does not require a change until after that point, of course, but just a note for later.